addons/product/static/src/js/product_configurator/product_template_attribute_line/product_template_attribute_line.js

```diff
@@ -0,0 +1,135 @@
+
+    /**
+     * Update the selected PTAV in the state.
+     *
+     * @param {Event} event
+     */
+    updateSelectedPTAV(event) {
+        this.env.updateProductTemplateSelectedPTAV(
+            this.props.productTmplId, this.props.id, event.target.value, this.props.attribute.display_type == 'multi'
+        );
+    }
+
+    /**
+     * Update in the state the custom value of the selected PTAV.
+     *
+     * @param {Event} event
+     */
+    updateCustomValue(event) {
+        this.env.updatePTAVCustomValue(
+            this.props.productTmplId, this.props.selected_attribute_value_ids[0], event.target.value
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return template name to use by checking the display type in the props.
+     *
+     * Each attribute line can have one of this five display types:
+     *      - 'Color'  : Display each attribute as a circle filled with said color.
+     *      - 'Pills'  : Display each attribute as a rectangle-shaped element.
+     *      - 'Radio'  : Display each attribute as a radio element.
+     *      - 'Select' : Display each attribute in a selection tag.
+     *      - 'Multi'  : Display each attribute in a multi-checkbox tag.
+     *
+     * @return {String} - The template name to use.
+     */
+    getPTAVTemplate() {
+        switch(this.props.attribute.display_type) {
+            case 'color':
+                return 'product.ptav-color';
+            case 'multi':
+                return 'product.ptav-multi';
+            case 'pills':
+                return 'product.ptav-pills';
+            case 'radio':
+                return 'product.ptav-radio';
+            case 'select':
+                return 'product.ptav-select';
+        }
+    }
+
+    /**
+     * Return the name of the PTAV
+     *
+     * In the selection HTML tag, it is impossible to show the component `BadgeExtraPrice`. Append
+     * the extra price to the name to ensure that the extra price will be shown.
+     * Note: used in `productConfigurator.ptav-select`.
+     *
+     * @param {Object} ptav - The attribute, as a `product.template.attribute.value` summary dict.
+     * @return {String} - The name of the PTAV.
+     */
+    getPTAVSelectName(ptav) {
+        return ptav.name;
+    }
+
+    /**
+     * Check if the selected ptav is custom or not.
+     *
+     * @return {Boolean} - Whether the selected ptav is custom or not.
+     */
+    isSelectedPTAVCustom() {
+        return this.props.attribute_values.find(
+            ptav => this.props.selected_attribute_value_ids.includes(ptav.id)
+        )?.is_custom;
+    }
+
+    /**
+     * Check if there are only custom ptav or not.
+     *
+     * @return {Boolean} - Whether all the ptav are custom.
+     */
+    isAllPTAVCustom() {
```

Comment:
Why is that event needed? Don't you have a parameter `remove_custom` in `_loadData` ? 
```suggestion
    areAllPTAVCustom() {
```